# ============================ filter_improved.yaml ============================
self_filter:
  ros__parameters:
    sensor_frame: "Lidar"

    # For spheres (single scale & padding)
    default_sphere_scale: 1.0
    default_sphere_padding: 0.01

    # For boxes (x,y,z scale & padding)
    default_box_scale: [1.0, 1.0, 1.0]
    default_box_padding: [0.01, 0.01, 0.01]

    # For cylinders (radial scale, vertical scale, radial padding, vertical padding)
    default_cylinder_scale: [1.0, 1.0]
    default_cylinder_padding: [0.01, 0.01]

    keep_organized: false
    zero_for_removed_points: false
    invert: false
    min_sensor_dist: 2.0

    # Link definitions
    self_see_links:
      names:
        - BASE
        - CABIN
        - ROTO
        - ROTO_BASE
        - ENDEFFECTOR
        - SHOVEL
        - STICK
        - TELE
        - BOOM
        - LF_WHEEL
        - LH_WHEEL
        - RF_WHEEL
        - RH_WHEEL

      # Example custom overrides:
      TELE:
        # If TELE is a box, define 3 scale factors and 3 paddings
        box_scale: [1.0, 1.0, 1.0]
        box_padding: [0.05, 0.05, 0.05]
      
      STICK:
        box_scale: [1.1, 1.1, 1.7]
        box_padding: [0.05, 0.22, 0.05]
      
      BOOM:
        # If BOOM is a box, define 3 scale factors and 3 paddings
        box_scale: [1.0, 1.0, 1.3]
        box_padding: [0.05, 0.1, 0.22]

      LF_WHEEL:
        # If wheels are cylinders: radial scale, vertical scale, radial padding, vertical padding
        cylinder_scale: [1.0, 1.0]
        cylinder_padding: [0.05, 0.3]

      LH_WHEEL:
        cylinder_scale: [1.0, 1.0]
        cylinder_padding: [0.05, 0.3]

      RF_WHEEL:
        cylinder_scale: [1.0, 1.0]
        cylinder_padding: [0.05, 0.3]

      RH_WHEEL:
        cylinder_scale: [1.0, 1.0]
        cylinder_padding: [0.05, 0.3]
